components/bl-mapbox-component/src/helpers.js

```diff
@@ -121,64 +121,77 @@ export const useMarkers = (markers, markersArray, setMarkersArray, mapRef, onMar
   }, [markers]);
 };
 
-export const usePolygons = (polygons, polygonsArray, setPolygonsArray, mapRef, onPolygonClick) => {
-  useEffect(() => {
-    if (polygons?.length && mapRef.current) {
-      mapRef.current.on('load', () => {
-        polygonsArray.forEach(polygon => {
-          if (mapRef.current.getSource(polygon)) {
-            mapRef.current.removeLayer(`${ polygon.id }-layer`);
-            mapRef.current.removeSource(polygon.id);
-          }
-        });
+const updatePolygonsArray = (polygons, mapRef, polygonsArray, setPolygonsArray) => {
+  if (polygons?.length && mapRef.current) {
+    mapRef.current.on('load', () => {
+      polygonsArray.forEach(polygon => {
+        if (mapRef.current.getSource(polygon)) {
+          mapRef.current.removeLayer(`${ polygon.id }-layer`);
+          mapRef.current.removeSource(polygon.id);
+        }
```

Comment:
Can we check if the polygon changed and remove then only if it is?

Reply:
We are already checking it in usePolygons function.